inc/cleantalk-common.php

```diff
@@ -516,6 +518,29 @@ function apbct_sender_info___get_page_url()
     return  Server::get('SERVER_NAME') . Server::get('REQUEST_URI');
 }
 
+function apbct_visible_fields__get_collection()
+{
+    global $apbct;
+
+    if( $apbct->settings['data__set_cookies'] == 1 ) {
+        // Get from separated native cookies and convert it to collection
+        $visible_fields_cookies_array = array_filter( $_COOKIE, static function ( $key ) {
+            return strpos($key, 'apbct_visible_fields_') !== false;
+        }, ARRAY_FILTER_USE_KEY );
+        $visible_fields_collection = array();
+        foreach ( $visible_fields_cookies_array as $visible_fields_key => $visible_fields_value ) {
+            $prepared_key = str_replace('apbct_visible_fields_', '', $visible_fields_key);
+            $prepared_value = json_decode(str_replace('\\', '', $visible_fields_value), JSON_OBJECT_AS_ARRAY);
+            $visible_fields_collection[$prepared_key] = $prepared_value;
+        }
+    } else {
+        // Get from alt cookies storage
+        $visible_fields_collection = Cookie::get('apbct_visible_fields', array(), 'array');
+    }
+
+    return $visible_fields_collection;
+}
+
```

Comment:
Cookie::get() already has all the code to differ with which type of cookie we are working with ` $apbct->settings['data__set_cookies']`

Shouldn't we better extend the method Cookie::get() instead of making another layer like apbct_visible_fields__get_collection()?

Reply:
Moved https://github.com/CleanTalk/wordpress-antispam/commit/8a4f735c572cb2a16160ac1f94bc58c82c19e719